### PR TITLE
Refactor TOC to support anchor specified in metadata

### LINKED ISSUE #100: Refactor TOC to support anchor specified in metadata
Current TOC(flat) generates anchor links by sections under the main div automatically. In Brand Detail pages, there are multiple sections that don't match with TOC anchors. Refactor TOC to support anchor specified in metadata, such as 
image

### PR DESCRIPTION
**Pull request by JiangLong2019**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #100 

Test URLs:
- Original: https://www.sunstar.com/brands/consumer-health-beauty/
- Before: https://main--sunstar--hlxsites.hlx.live/brands/consumer-health-beauty
- After: https://issue-100--sunstar--hlxsites.hlx.live/brands/consumer-health-beauty

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://<branch>--sunstar--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/<your-block>&index=0

### CODE REVIEW

File: blocks/toc/toc.js [hunk 1]
```diff
@@ -52,25 +52,18 @@ function buildTOCSide(ul, block) {
...
-        liItem.innerText = '';
-        liItem.append(aLink);
-      }
-    }
-    if (section.classList.contains('toc-container')) {
-      others = true;
-      tocIndex = i;
-    }
+
+  [...ul.querySelectorAll('li')].forEach((li) => {
```
Comment (dnbute) [inline blocks/toc/toc.js:56]: Could this also be implemented for the side TOC as well to maintain consistency?

Comment (JiangLong2019) [reply blocks/toc/toc.js:56]: @dnbute Unfortunately I don't have this needs for current page. If you have we can do it, or we can do it in the future.

File: blocks/toc/toc.js [hunk 2]
```diff
@@ -52,25 +52,18 @@ function buildTOCSide(ul, block) {
...
+    const liVar = li.innerText.toLowerCase();
+    const aLink = document.createElement('a');
+    aLink.href = `#${liVar}`;
+    aLink.innerText = li.innerText;
+    li.innerText = '';
+    li.append(aLink);
+  });
+  [...document.querySelectorAll('[data-toc]')].forEach((section) => {
```
Comment (sdmcraft) [inline blocks/toc/toc.js:64]: Nit: Maybe I'd rename the property `toc` to `toc-anchor` to convey the meaning.

**Review by sdmcraft — APPROVED** [review]

**Comment by sdmcraft** [comment]
It'd be great if you can add an example for this in the block library. Please ignore if done already.

**Comment by JiangLong2019** [comment]
All done

**Pull request merged**